common/core/web/keyboard-processor/src/text/defaultOutput.ts

```diff
@@ -24,17 +24,17 @@ namespace com.keyman.text {
      * Serves as a default keycode lookup table.  This may be referenced safely by mnemonic handling without fear of side-effects.
      * Also used by Processor.defaultRuleBehavior to generate output after filtering for special cases.
      */
-    public static forAny(Lkc: KeyEvent, isMnemonic: boolean) {
+    public static forAny(Lkc: KeyEvent, isMnemonic: boolean, ruleBehavior?: RuleBehavior) {
       var char = '';
 
       // A pretty simple table of lookups, corresponding VERY closely to the original defaultKeyOutput.
-      if((char = DefaultOutput.forSpecialEmulation(Lkc)) != null) {
+      if((char = DefaultOutput.forSpecialEmulation(Lkc, ruleBehavior)) != null) {
         return char;
-      } else if(!isMnemonic && ((char = DefaultOutput.forNumpadKeys(Lkc)) != null)) {
+      } else if(!isMnemonic && ((char = DefaultOutput.forNumpadKeys(Lkc, ruleBehavior)) != null)) {
         return char;
-      } else if((char = DefaultOutput.forUnicodeKeynames(Lkc)) != null) {
+      } else if((char = DefaultOutput.forUnicodeKeynames(Lkc, ruleBehavior)) != null) {
         return char;
-      } else if((char = DefaultOutput.forBaseKeys(Lkc)) != null) {
+      } else if((char = DefaultOutput.forBaseKeys(Lkc, ruleBehavior)) != null) {
```

Comment:
The latter two already (optionally) took the second parameter, allowing the method to return warnings for select scenarios.  The problem was... we never actually provided that parameter in 14.0.  Oops!